manage/ships/edit.php

```diff
@@ -0,0 +1,147 @@
+<?php 
+// Display already entered data and allow uses to edit it. Doing similar checks to new
+
+require '../layout/header.php';
+
+use Model\AbstractShip;
+use Service\Container;
+
+$container = new Container($configuration);
+$shipLoader = $container->getShipLoader();
+$teams = AbstractShip::getTeams();
+$errors = [];
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    // set the shipID and instantiate a ship
+    if (isset($_POST['shipId'])) {
+        $id = $_POST['shipId'];
+        $ship = $shipLoader->findOneById($id);
+        if ($ship == null) {
+            echo 'SHIP NOT FOUND';
+            die;
+        }
+    } else {
+        throw new \Exception('NO SHIP WITH THIS ID FOUND');
+        // or could catch and die the error
+    }
+
+    $name = $_POST['name'];
```

Comment:
I think the logic for all of these checks can change slightly now. You can go ahead and assign the variables to the ship object instead of setting to a variable and then assigning below to the object.
```
$ship->setName($_POST['name']);
if (empty($ship->getName())) {
  $errors[] = '...';
}
```

This also helps out when you try to submit the form with bad data. Currently if you do that, it will not show the data you tried to submit in the form, but the original data that is in the `$ship` object.

Let me know if you want to discuss this more.

Reply:
Done